Content.Server/Objectives/Conditions/HijackShuttleCondition.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Content.Server.Mind;
+using Content.Server.Roles;
+using Content.Server.Shuttles.Components;
+using Content.Shared.Cuffs.Components;
+using Content.Shared.Humanoid;
+using Content.Shared.Mind;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Objectives.Interfaces;
+using Robust.Shared.Player;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Objectives.Conditions;
+
+[DataDefinition]
+public sealed partial class HijackShuttleCondition : IObjectiveCondition
+{
+    private MindComponent _mind;
+    private EntityUid _mindId;
+
+    public IObjectiveCondition GetAssigned(EntityUid mindId, MindComponent mind)
+    {
+        return new HijackShuttleCondition
+        {
+            _mind = mind,
+            _mindId = mindId,
+        };
+    }
+
+    public string Title => Loc.GetString("objective-condition-hijack-shuttle-title");
+
+    public string Description => Loc.GetString("objective-condition-hijack-shuttle-description");
+
+    public SpriteSpecifier Icon => new SpriteSpecifier.Rsi(new ResPath("Objects/Tools/emag.rsi"), "icon");
+
+    private bool IsShuttleHijacked(EntityUid shuttleGridId)
+    {
+        var entMan = IoCManager.Resolve<IEntityManager>();
+        var mindSystem = entMan.System<MindSystem>();
+        var roleSystem = entMan.System<RoleSystem>();
+        var mobStateSystem = entMan.System<MobStateSystem>();
+
+        var agentOnShuttle = false;
+        var gridPlayers = Filter.BroadcastGrid(shuttleGridId).Recipients;
+        var humanoids = entMan.GetEntityQuery<HumanoidAppearanceComponent>();
+        var cuffable = entMan.GetEntityQuery<CuffableComponent>();
+        entMan.EntityQuery<MobStateComponent>();
+        foreach (var player in gridPlayers)
+        {
+            if (!player.AttachedEntity.HasValue ||
+                !mindSystem.TryGetMind(player.AttachedEntity.Value, out var mindId, out _))
+                continue;
+
+            if (mindId == _mindId)
+            {
+                agentOnShuttle = true;
+                continue;
+            }
+
+            var isHumanoid = humanoids.HasComponent(player.AttachedEntity.Value);
+            if (!isHumanoid) // Only humanoids count as enemies
+                continue;
+
+            var isPersonTraitor = roleSystem.MindHasRole<TraitorRoleComponent>(mindId);
```

Comment:
theres IsAntagonist or something which uses an event and is more flexible - if a traitor has to hijack shuttle a loneop or ninja shouldnt stop that
the description also directly mentions nanotrasen enemies

Reply:
Checking antag role comp will be enough? https://github.com/space-wizards/space-station-14/pull/19621/commits/2acc86b2175db34becb9add75824343001e7a0c0

Reply:
hardcoding components isnt ideal, use `SharedRoleSystem`'s `MindIsAntagonist(mindId)`

Reply:
Done